workers/unity/Assets/Fps/Scripts/Config/MapBuilder.cs

```diff
@@ -81,45 +84,44 @@ public IEnumerator CleanAndBuild(int worldLayers = 4, string seed = "SpatialOS G
             InitializeGroupsAndComponents();
 
             // Initialize tiles
-            SceneManager.SetActiveScene(tempScene);
-            foreach (var tileCollection in mapTemplate.tileCollections)
+            using (new SceneScope(tempScene))
             {
-                tileCollection.LoadAndOptimizeTiles();
-            }
-
-            mapTemplate.defaultTileCollection.LoadAndOptimizeTiles();
-            SceneManager.SetActiveScene(activeScene);
+                foreach (var tileCollection in mapTemplate.tileCollections)
+                {
+                    tileCollection.LoadAndOptimizeTiles();
+                }
 
-            yield return null;
+                mapTemplate.defaultTileCollection.LoadAndOptimizeTiles();
```

Comment:
why did that `yield return null;` go away

Reply:
This was an unneeded frame wait, removing it allowed me to simplify the code significantly

Reply:
is it also unneeded on mobile :D cause I don't want it to look like it's hanging